db/sql.h

```diff
@@ -79,8 +79,13 @@ typedef struct stmt_hash_entry {
 } stmt_hash_entry_type;
 
 struct sql_authorizer_state {
-    int denyDdl;                       /* non-zero if DDL is forbidden */
-    int numDdls;                       /* number of DDL statements found */
+    struct sqlclntstate *clnt;         /* pointer to current client info */
+    int denyDdl;                       /* DDL is forbidden? */
+    int denyPragma;                    /* PRAGMA is forbidden? */
+    int denyTrigger;                   /* CREATE TRIGGER is forbidden? */
```

Comment:
Move them into a single 'flag' variable with OR-able flags?

Reply:
Good point.  I actually thought about doing that; however, I was afraid it would hurt the readability too much.  I suppose the code will be easier to maintain with just the flags versus having individual boolean fields.

Reply:
I've made the necessary changes.  While making the changes, I discovered an issue with the authorizer integration that I also fixed (i.e. it assumed that "thd" was always non-NULL).